### ISSUE TITLE
No way to fix the result(random_state or random_seed)

### ISSUE THREAD

**Issue by ryona-noda** [pfnet/optuna#469]
Searching the documentation and API shows no way to fix the result

example:
`
study = optuna.create_study();
study.optimize(opt,random_state=123)
`

**Comment by sile** [comment]
Thank you for your question. 
I think that the QA entry ["How can I obtain reproducible optimization results?"](https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-obtain-reproducible-optimization-results) will help you.

**Comment by ryona-noda** [comment]
Thank you very much for your kind support!
All resoleved!!

**Comment by ryona-noda** [comment]
Sorry , The problem is not solved yet.
Please have a look at my code...
```
import optuna
from sklearn.metrics import mean_squared_error

def opt(trial):
    n_estimators = trial.suggest_int('n_estimators', 0, 250)
    xgboost_tuna = xgb.XGBRegressor(
        random_state=1234,
        n_estimators = n_estimators,
    )
    xgboost_tuna.fit(train_df_x,train_df_y)
    tuna_pred_train = xgboost_tuna.predict(train_df_x)
    return mean_squared_error(train_df_y,tuna_pred_train)

study = optuna.create_study(sampler=optuna.samplers.TPESampler(seed=10))
study.optimize(opt)

print(study.best_params)
```

**Comment by sile** [comment]
I cannot reproduce your problem because the above code failed with the error `NameError("name 'train_df_x' is not defined",)`.
So, I'm not sure, but if you used [`sklearn.model_selection.train_test_split`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) to split your data, specifying a random seed to the function may resolve your problem.
(cf. [How to set the global random_state in Scikit Learn](https://www.mikulskibartosz.name/how-to-set-the-global-random_state-in-scikit-learn/))

**Comment by ryona-noda** [comment]
Sorry , This is full code.

```
# 諸々読み込み
from sklearn.metrics import mean_squared_error
from sklearn import model_selection
from sklearn import datasets
import pandas as pd
import xgboost as xgb
import optuna

# ハイパーパラメータ探索用関数
def opt(trial):
    n_estimators = trial.suggest_int('n_estimators', 0, 250)
    max_depth = trial.suggest_int('max_depth', 1, 5)
    min_child_weight = trial.suggest_int('min_child_weight', 1, 15)
    subsample = trial.suggest_int('subsample', 1, 10)*0.1
    colsample_bytree = trial.suggest_int('colsample_bytree', 1,10)*0.1
    importance_type = trial.suggest_categorical('importance_type', ['weight', 'gain', 'cover'])
    tree_method = trial.suggest_categorical('tree_method', ['auto','exact','approx','hist'])
    grow_policy = trial.suggest_categorical('grow_policy', ['depthwise','lossguide'])
    xgboost_tuna = xgb.XGBRegressor(
        random_state=1234,
        n_estimators = n_estimators,
        max_depth = max_depth,
        min_child_weight = min_child_weight,
        subsample = subsample,
        colsample_bytree = colsample_bytree,
        importance_type = importance_type,
        tree_method = tree_method,
        grow_policy = grow_policy,
    )
    xgboost_tuna.fit(train_df_x,train_df_y)
    tuna_pred_train = xgboost_tuna.predict(train_df_x)
    return mean_squared_error(train_df_y,tuna_pred_train)

# データ読み込み
boston = datasets.load_boston()
boston_df = pd.DataFrame(boston.data)
boston_df['target'] = boston.target

# 訓練データとテストデータに分割
train_df, test_df = model_selection.train_test_split(boston_df, test_size=0.2,random_state=0)
train_df_y = train_df[['target']]
train_df_x = train_df.copy().drop('target', axis=1)
test_df_y = test_df[['target']]
test_df_x = test_df.copy().drop('target', axis=1)

# 最適化
study = optuna.create_study(sampler=optuna.samplers.TPESampler(seed=10))
study.optimize(opt, n_trials=100,n_jobs=-1)

# 結果出力
print(study.best_params)
```

**Comment by ryona-noda** [comment]
Probably, This part is the cause...

```
subsample = trial.suggest_int('subsample', 1, 10)*0.1
colsample_bytree = trial.suggest_int('colsample_bytree', 1,10)*0.1
```
&
```
subsample = subsample,
colsample_bytree = colsample_bytree
```

**Comment by ryona-noda** [comment]
Maybe, caused by 'n_jobs=-1'??

**Comment by sile** [comment]
> Maybe, caused by `n_jobs=-1`??

Yes, I think so.
If `n_jobs=-1` is specified and your machine has multiple CPU cores, the optimization is executed in parallel.

The following sentence is quoted from [How can I obtain reproducible optimization results?](https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-obtain-reproducible-optimization-results):
> First, when optimizing a study in distributed or parallel mode, there is inherent non-determinism. 
 Thus it is very difficult to reproduce the same results in such condition. We recommend executing optimization of a study sequentially if you would like to reproduce the result.

**Comment by ryona-noda** [comment]
Thank you very much for your kind support!
All resoleved!!
